### PR TITLE
Armless Cuffing and Legless Legcuffing Sense-Making

### LINKED ISSUE #4873: You cannot escape justice. (you can cuff people with one or literally no arms)
**Problem Description**:
You can cuff people with one or literally no arms at all.

**What did you expect to happen**: 
GEE I DON'T KNOW img
HE SURE HAS A LOT TO CUFF THERE DOESN'T HE

**What happened instead**:
OH MY GOD WHAT DO YOU THINK

**Why is this bad/What are the consequences:**
WELL GEE I SURE LIKE CUFFING MY AMPUTEE SPACEMENS. IT MAKES NO SENSE WHAT

**Steps to reproduce the problem**:
CHOP LIMB(S) PUT CUFFS ON. HAND/ARM MISSING DON'T MAKE A DIFFERNCE

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: WHO MADE THIS TEMPLATE AGH

### PR DESCRIPTION
**Pull request by TullyBurnalot**
Fixes #4873

This was entirely DZD's work. It has my name on it because it was a "learn-as-you-code" experience.

Regardless:

- Can now cuff people with two hands, or one hand;
- Can no longer cuff people with no hands
- Can no longer try to cuff non-carbon mobs and make the game freak out

:cl: TullyBBurnalot
tweak: Handless cuffing no longer possible. Can still cuff people with one hand
/:cl:

### CODE REVIEW

**Comment by monster860** [comment]
You shouldn't be able to cuff one-handed people.

**Comment by TullyBurnalot** [comment]
You can still cuff one of their hands and drag them by it. Assume it's getting cuffed to their neck or something.

**Comment by TullyBurnalot** [comment]
Also note, you can, for instance, cuff a one-handed person to a pipe

**Comment by Plotron** [comment]
@monster860 - we don't want that to be abused. You can shoot people with one hand just fine... why would you be exempt from being handcuffed?

**Comment by Regen1** [comment]
technically there are handcuffs made for people with one hand
We're in space, they work it out

**Comment by Regen1** [comment]
Also, please add the ability to cuff legs

Target legs / feet = leg cuffs, target can only walk

**Comment by monster860** [comment]
> target can only walk

How about additionally, a chance of falling over? Because you're jumping around?

**Comment by Ar3nn** [comment]
They have that, you use zip-ties on shoes to make them for some godawful reason.

**Comment by TullyBurnalot** [comment]
@Ar3nn Not shoes, prison orange shoes. VERY SPECIFICALLY prison orange shoes.

**Comment by KasparoVy** [comment]
that's wild

**Comment by Krausus** [comment]
This doesn't stop security bots from zip-tying handless people, though. Which, logically, will lead to every officer dragging around a securitron, just in case some armless hooligan shows up.

**Comment by Spacemanspark** [comment]
I doubt that considering that beepsky can already do this now and nobody does it.

**Comment by Crazylemon64** [comment]
It's a feature, Beepsky is an interdimensional robocop that exists outside of time

**Comment by Davdichan** [comment]
If we are going realism a pair of handcuffs or zipties aren't going to stop a person who wants to resist from fighting back by kicking or just moving the cuffs under their legs and hammering/choking you with them.

**Comment by Plotron** [comment]
@Davdichan: "Cuffed man chokes and hammers an officer"

**Comment by theColdflame** [comment]
:+1:  you could just cuff the remaining arm to the target's leg if they're one-armed, logically.

**Comment by FreeStylaLT** [comment]
how the fuck would that work cold

are all spacemen anorexic enough to have thighs big enough for a handcuff designed for wrists

**Comment by theColdflame** [comment]
if the sprites are to be believed

yes

**Comment by FreeStylaLT** [comment]
fucking raisin diets

**Comment by TullyBurnalot** [comment]
As I noted in the first post, I fixed legcuffs and allowed them to work again.

Then I added a box of them to the Warden's locker.

Legcuffs are applied to people the same way as handcuffs (targetting the legs not required). When succesfully applied, they force the person to walk until the legcuffs are removed, or they can break free of the legcuffs (resisting takes 30 seconds).

**Comment by FreeStylaLT** [comment]
Can you legcuff people with just one leg?

**Comment by TullyBurnalot** [comment]
Nope. You can only legcuff people with both legs.

**Comment by Fox-McCloud** [comment]
We don't need legcuffs *and* handcuffs.

We have shackles on shoes that functionally fit this purpose already. No need to copy-paste a ton of code and create a lot of new edge cases.

**Comment by TullyBurnalot** [comment]
The current shoe shackles work on one type of shoe, require zipties anyway, and need to be forced onto a prisoner's inventory.

This is a simpler solution.

**Comment by Fox-McCloud** [comment]
You literally copy-pasted handcuff code for leg cuffs. That's a "huh-uh".

Also, it's seriously not that hard to get a pair of orange shoes and handcuffs. There's what? 3-4 lockers each with 3 pairs.

It's not that hard to grab a pair of those and a pair of handcuffs.

**Comment by TullyBurnalot** [comment]
'Twas not arguing this was anything but cannibalized code.

I was arguing from an in-game standpoint, it's vastly more convenient to just apply legcuffs than it is to to get zipties, apply to shoes, then apply shoes.

File: code/game/objects/items/weapons/legcuffs.dm [hunk 1]
```diff
@@ -6,7 +6,7 @@
 	icon_state = "handcuff"
 	flags = CONDUCT
 	throwforce = 0
-	w_class = 3.0
+	w_class = 2.0
```
Comment (tigercat2000) [inline code/game/objects/items/weapons/legcuffs.dm]: remove the `.0` :tiger:

**Comment by CrunkyBars** [comment]
Apparently, what you do when you handcuff a one-handed person, is usually attach the other cuff to the person's belt, though I don't know whether the NT jumpsuits even have belt loops, let alone belts.

Another method that people seem to use when that's not an option, is knock the person down and cuff their existing wrist to their ankle, kinda like hogtying, but I don't know how hard that'd be to implement.

**Comment by TullyBurnalot** [comment]
Applying handcuffing AND legcuffing at the same time because someone has a single hand would be a tad bit overkill.

**Comment by Fox-McCloud** [comment]
I'm ok with the fixes, but not with the legcuffs. You can't resist out of legcuffs, it's copy-pasted code, and you can already do this with orange shoes+handcuffs.

**Comment by TullyBurnalot** [comment]
@Fox-McCloud I tested the changes, and you **can** actually resist out of the legcuffs, even quicker than handcuffs, in fact.

**Comment by Fox-McCloud** [comment]
>And as I said, the fact that there's little to no indication that the zipties-on-shoes thing even exists, I have literally, literally never seen it used, ever.

Then add it to the crafting page somewhere--something already existing and functioning 100% fine yet being little used isn't an excuse to copy-pasta code like this.

**Comment by TullyBurnalot** [comment]
@Fox-McCloud Merge conflicts fixed, legcuffs also removed. Will consider alternative way of making zipties-on-shoes more open for use.

**Comment by Fox-McCloud** [comment]
@TullyBurnalot 

It could probably be a tablecrafting recipe.

**Comment by TullyBurnalot** [comment]
@Fox-McCloud Could stand to make it with every cuff type as well, not just zipties.

I mean, really, handcuffs would make better shoecuffs than zipties.

**Comment by Fox-McCloud** [comment]
>@Fox-McCloud Could stand to make it with every cuff type as well, not just zipties.

It does; ANY cuffs. The cuffs+orange shoes things is *ancient* Tully. Literally ancient. I'm pretty sure it has roots in the earliest days of TG.

**Comment by TullyBurnalot** [comment]
@Fox-McCloud We need better encyclopedias...

Oh well, not going to add anything else to this PR. Sanity check for non-carbon added, one-handed cuffing is now a feature.

**Pull request merged**